services/wallet/transfer/transaction_manager_multitransaction.go

```diff
@@ -439,28 +430,8 @@ func (tm *TransactionManager) GetBridgeDestinationMultiTransaction(ctx context.C
 	return nil, nil
 }
 
-func (tm *TransactionManager) getVerifiedWalletAccount(address, password string) (*account.SelectedExtKey, error) {
-	exists, err := tm.accountsDB.AddressExists(types.HexToAddress(address))
-	if err != nil {
-		log.Error("failed to query db for a given address", "address", address, "error", err)
-		return nil, err
-	}
-
-	if !exists {
-		log.Error("failed to get a selected account", "err", transactions.ErrInvalidTxSender)
-		return nil, transactions.ErrAccountDoesntExist
-	}
-
-	key, err := tm.gethManager.VerifyAccountPassword(tm.config.KeyStoreDir, address, password)
-	if err != nil {
-		log.Error("failed to verify account", "account", address, "error", err)
-		return nil, err
-	}
-
-	return &account.SelectedExtKey{
-		Address:    key.Address,
-		AccountKey: key,
-	}, nil
+func generateMultiTransactionID() wallet_common.MultiTransactionIDType {
+	return wallet_common.MultiTransactionIDType(rand.Uint32())
```

Comment:
>  The worst that can happen imo, is that once in 9 billion times, the generated ID will be the same, we will fail to insert the transfer to DB and fail before sending the transaction, but not breaking the data integrity. No harm, though not awesome, user will be able to send the transaction with the next attempt.

How I consider a robust implementation should roughly do is by having the following steps:

- We open a `SQLite transaction` that will be reverted if all the blockchain calls fails
- We use the `SQLite transaction` to record all the persistance we expect to have: multi transaction entry, pending transaction entry and the relation between them
- We publish the sub-transactions and commit the `SQLite transaction` in case of no error

I'm no even sure we need to spill out the ID used in the relation between multi-transaction and pending transactions outside persistance.

@saledjenic @dlipicar what do you think about this?

Reply:
> I think blocking the DB can cause many issues, but the achieved effect is arguable - basically being able to match multiple transfers into a single activity entry to display it for user. If the effect was critical, I'd say "yes" to this, but since the effect is not critical, I don't think we should use this approach, which again makes the code highly coupled, thus harder to maintain and less testable imo.

Indeed that is the ideal case and we have to do the right tradeoffs for an efficient implementation as you mentioned.

Reply:
> The worst that can happen imo, is that once in some billion times, the generated ID will be the same, we will fail to insert the transfer to DB and fail before sending the transaction, but not breaking the data integrity.

Inserting into the db should come after submitting a tx.

When calling `generateMultiTransactionID` just to be safe maybe we can check if the generated uid exists in the db or not?

> But at the same time, the only good thing about previous implementation was that it that it was unique but on a specific device, unless we try to sync this between devices, and there is still a chance though not more that 1 in 9 Billion for the value to overflow, almost just like with the new implementation.

Users don't have the option to sync devices later, like running the app for the same Status profile on two devices for some time and then deciding to pair them. The only way to have paired devices is at the moment of creating a Status profile on other device by syncing from an already set up device, after creating process is done the state on both devices is the same.